tests/test_gmail_utils.py

```diff
@@ -0,0 +1,96 @@
+"""
+This is an example automated test to check gmail utils
+Our automated test will do the following:
+    #login to gmail and fetch mailboxes
+    #After fetching the mail box ,select and fetch messages and print the number of messages and the subject of the messages
+
+Prerequisites:
+    - Gmail account with app password
+"""
+
+import sys
+import os
+from integrations.reporting_channels.gmail.gmail import Gmail
+from integrations.reporting_channels.gmail.mailbox import Mailbox
+from dotenv import load_dotenv
+sys.path.append(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
+import pytest
+
+load_dotenv()
+
+@pytest.mark.GMAIL
+def test_gmail_util():    
+    try:
+        # Initialize flags for test summary
+        expected_pass = 0
+        actual_pass = 0
+
+        # Initialize Gmail class instance
+        gmail = Gmail()
+        gmail.connect()
+
+        username = os.getenv('app_username')
+        password = os.getenv('app_password')
+
+        # Attempt to log in
+        result_flag = gmail.login(username, password)
+        assert result_flag, "Login failed!"
+        print("Login successful!")
+        expected_pass += 1
+        actual_pass += 1
+
+        # Fetch and print mailboxes
+        mailboxes = gmail.fetch_mailboxes()
+        assert mailboxes, "Failed to fetch mailboxes!"
+        print(f"Fetched mailboxes: {mailboxes}")
+        expected_pass += 1
+        actual_pass += 1
+
+        # Select the SPAM mailbox
+        inbox_mailbox = gmail.use_mailbox('[Gmail]/Spam')

```

Comment:
We are just pulling messages from spam, can this be configurable?

Reply:
Since this is a test script, I have used spam folder to pull and to check for further email utilities function, it can be passed by the conf file and also the gmail.py has a separate function to pull from Spam,
So here to use the spam is intentional since I am testing the gmail.use_mailbox functionality by calling it from the test script.

Reply:
Though its a test script this script is going to be used by others. Make this as a variable which can be passed as an argument. This way users dont need to update the code to run it for a different mail box. You can default it to spam